### ISSUE TITLE
Generate Unit Tests for github_service.py

### ISSUE THREAD

**Issue by dhirenmathur** [potpie-ai/potpie#175]
**Description:**

We need to create unit tests for the `github_service.py` file located in the `app/modules/github/` directory. The goal is to ensure that all functions are thoroughly tested for expected behavior, edge cases, and error handling.

**Functions to Test:**

1. **initialize_tokens**: Ensure that tokens are initialized correctly from environment variables and handle cases where the token list is empty.
2. **get_github_repo_details**: Test the retrieval of GitHub repository details, including handling of invalid repository names and API errors.
3. **get_file_content**: Validate file content retrieval, including:
   - Successful access to private and public repositories.
   - Handling of directory paths instead of file paths.
   - Correct line selection based on `start_line` and `end_line`.
4. **_detect_encoding**: Test encoding detection, including cases with low confidence.
5. **get_github_oauth_token**: Ensure that the correct OAuth token is retrieved for a given user ID and handle cases where the user is not found.
6. **get_repos_for_user**: Validate the retrieval of repositories for a user, including error handling for missing users and tokens.
7. **get_combined_user_repos**: Test the combination of user repositories and project repositories, ensuring no duplicates.
8. **get_branch_list**: Validate branch retrieval, including error handling for non-existent repositories.
9. **get_public_github_instance**: Ensure that a public GitHub instance is returned correctly.
10. **get_repo**: Test the retrieval of a repository, including handling of both private and public access failures.
11. **get_project_structure_async**: Validate the fetching of project structure, including caching behavior.
12. **_fetch_repo_structure_async**: Test the recursive fetching of repository structure, including depth handling.
13. **_format_tree_structure**: Ensure correct formatting of the tree structure output.

**Testing Framework:**
- Use `pytest` for writing the unit tests.
- Consider using `pytest-mock` or `unittest.mock` to mock external dependencies like GitHub API calls and database interactions.

**Acceptance Criteria:**
- All functions have corresponding unit tests.
- Tests cover both happy paths and edge cases.
- Code coverage should be at least 80%.

**Additional Notes:**
- Please ensure that the tests are organized and follow best practices for readability and maintainability.

**Comment by dhirenmathur** [comment]
@Anu-Ra-g that's great, you should create a new folder in the root directory of the project called `tests`

**Comment by Anu-Ra-g** [comment]
I'm using this for setting up the `tests` package, instead of using `__init__.py`
```
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../')))

from app.modules.github.github_service import GithubService
```
Will it okay?

**Comment by dhirenmathur** [comment]
@Anu-Ra-g I would recommend to stick to creating the init file, this approach could end up being difficult to maintain and run tests from other directories

**Comment by dhirenmathur** [comment]
@Anu-Ra-g let us know if you need any help

**Comment by dhirenmathur** [comment]
@Anu-Ra-g are you still working on this?

**Comment by Anu-Ra-g** [comment]
Yes, I'm still working on this but I got a little busy. I will make the PR as soon as possible.

**Comment by Anu-Ra-g** [comment]
Go ahead

**Comment by simin75simin** [comment]
@dhirenmathur I'd like to solve this one as well if that is okay.

**Comment by dhirenmathur** [comment]
@simin75simin let's complete the other one first, there's already an open PR for this one too

**Comment by simin75simin** [comment]
> [@simin75simin](https://github.com/simin75simin) let's do the issues one at a time so that others have a chance as well, there's already an open PR for this one too :)

sorry just looking for things to contribute and this one seemed dormant

**Comment by dhirenmathur** [comment]
You are right that this is dormant @simin75simin, if we can validate that no work is needed in the other issue, you can pick this up! Thank you!

**Comment by Phantom0110** [comment]
Could I try working on this?

**Comment by Dsantra92** [comment]
duplicate of https://github.com/potpie-ai/potpie/issues/350
